homeassistant/components/zha/__init__.py

```diff
@@ -194,8 +261,11 @@ def device_removed(self, device):
             component = None
             profile_clusters = ([], [])
             device_key = "{}-{}".format(device.ieee, endpoint_id)
-            node_config = self._config[DOMAIN][CONF_DEVICE_CONFIG].get(
-                device_key, {})
+            node_config = {}
+            if CONF_DEVICE_CONFIG in self._config_entry.data:
+                node_config = self._config_entry.data[CONF_DEVICE_CONFIG].get(
```

Comment:
I think the `device_config` section should stay in configuration.yaml and not be imported into the config entry. There's no way to add/edit them in the config entry currently aside from the initial configuration.yaml import, and I believe the desire currently is to keep the more advanced options part of configuration.yaml

Reply:
Although not very intuitive one can delete the config entry and recreate it from configuration.yaml. I think the deconz component uses a similar logic.

Reply:
@cgarwood the `device_config` is only an option in the configuration.yaml and not part of the user step in the config flow. So this looks good!